client/overview/task-list/dispute-tasks.ts

```diff
@@ -0,0 +1,221 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import moment from 'moment';
+import { getHistory } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import type { TaskItemProps } from './types';
+import type { CachedDispute } from 'wcpay/types/disputes';
+import { formatCurrency } from 'wcpay/utils/currency';
+import { getAdminUrl } from 'wcpay/utils';
+import wcpayTracks from 'wcpay/tracks';
+
+/**
+ * Returns true if the dispute is due within the specified number of days.
+ *
+ * @param {CachedDispute} dispute - The dispute to check.
+ * @param {number} [days] - The number of days to check. Defaults to 1.
+ *
+ * @return {boolean} True if the dispute is due within the specified number of days.
+ */
+const isDueWithin = ( dispute: CachedDispute, days = 1 ) => {
+	const now = moment();
+	const dueBy = moment( dispute.due_by );
+	return (
+		dueBy.diff( now, 'hours' ) > 0 &&
+		dueBy.diff( now, 'hours' ) <= 24 * days
+	);
+};
+
+/**
+ * Returns an array of disputes that are due within the specified number of days.
+ *
+ * @param {CachedDispute[]} activeDisputes - The active disputes to filter.
+ * @param {number} days - The number of days to check.
+ *
+ * @return {CachedDispute[]} The disputes that are due within the specified number of days.
+ */
+export const getDisputesDueWithinDays = (
+	activeDisputes: CachedDispute[],
+	days: number
+): CachedDispute[] =>
+	activeDisputes.filter( ( dispute ) => isDueWithin( dispute, days ) );
+
+export const getDisputeResolutionTask = (
+	/**
+	 * Active disputes (awaiting a response) to generate the notice string for.
+	 */
+	activeDisputes: CachedDispute[]
+): TaskItemProps | null => {
+	const disputeCount = activeDisputes.length;
+
+	if ( disputeCount === 0 ) {
+		return null;
+	}
+
+	const handleClick = () => {
+		wcpayTracks.recordEvent( 'wcpay_overview_task', {
+			task: 'dispute-resolution-task',
+		} );
```

Comment:
Tracks considerations:
* Should we change this name to an action:  e.g. `wcpay_overview_task_click`?
* Should we include additional props: e.g. `activeDisputeCount`?

Reply:
🙏 Updated in 0f53fa85f3e4148eb31ede53278d91cea3b229e5

Reply:
🙏 Updated in 0f53fa85f3e4148eb31ede53278d91cea3b229e5